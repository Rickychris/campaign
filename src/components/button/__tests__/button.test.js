import React from "react";
import ReactDom from "react-dom";
import { render, cleanup } from "@testing-library/react";
import Button from "../button";

describe("Button tests", () => {
  beforeEach(cleanup);

  it("renders the button", () => {
    const div = document.createElement("div");
    ReactDom.render(<Button></Button>, div);
  });

  it("renders the button text correctly", () => {
    const { getByTestId } = render(<Button text="click me" />);
    expect(getByTestId("button")).toHaveTextContent("click me");
  });

  it("matches snapshot", () => {
    const { getByTestId } = render(<Button text="click me" />);
    expect(getByTestId("button")).toMatchSnapshot();
  });
});
