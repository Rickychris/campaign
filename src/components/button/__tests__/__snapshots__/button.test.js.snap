// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests matches snapshot 1`] = `
<button
  class="root"
  data-testid="button"
  text="click me"
>
  click me
</button>
`;
